public/app/features/explore/ExplorePage.tsx

```diff
@@ -54,7 +54,7 @@ export function ExplorePage(props: GrafanaRouteComponentProps<{}, ExploreQueryPa
 
   useEffect(() => {
     keybindings.setupTimeRangeBindings(false);
-  }, [keybindings]);
+  }, [keybindings, keybindings.bindings]);
```

Comment:
It's a bit obscure because to correctly implement binding someone writing the code needs to know to add `keybindings.bindings`. This is part of internal implementation of KeybindingSrv. It also makes it less abstract.

I'd prefer to just clean bindings before the render, e.g. move it to:

```typescript
  useLayoutEffect(() => {
    keybindings.clearAndInitGlobalBindings();
  }, []);
```

If that's not an option and there's no other way I'd suggest at least adding a custom hook `useKeybinding` that would encapsulate this logic.

Reply:
This is a much better solution - I reverted my changes and added this one. Thank you!